### DISCUSSION TITLE
Handle more than the latest commit?

### DISCUSSION THREAD

**Discussion by gotson** [gembaadvantage/uplift#137] [General]
Hi,

i am evaluating this tool as a replacement for `semantic-release`, but from the tests i've done, it seems it only looks at the latest commit, instead of all the commits that were added since the last release.

Is that on purpose? Can it be configured ?

Thanks

**Comment by ga-paul-t** [comment]
Hey. Apologies for missing your message. 

Thanks for taking a look at uplift. You are right. At the moment uplift was designed to integrate within your CI analysing the last commit message. So there is an assumption that you have used a squash merge strategy when merging PRs for example.

Would you mind describing your use case? It sounds like some functionality that would be worth implementing

Thanks

**Comment by gotson** [comment]
I found another project that fits more what i want to do: https://github.com/caarlos0/svu

**Reply by ga-paul-t** [reply]
Thanks for letting me know. I will take a look at that project and compare it to uplift

**Comment by tomphp** [comment]
+1 for more than the latest commit.

- the next version should be determined from all commits since the last release tag
- it should bump the version by whatever commit triggers the left-most change in the SemVer

Example use cases:

- developer pushes more that one commit
- PR is merged in with more than one commit
- PR is merged in with a merge commit (`Merge pull request #xxx` doesn't trigger a semver bump)

e.g.

- fix: xxx
- chore: yyy
- fix: zzz

would bump the patch version as there are no patches or breaking changes.

- fix: xxx
- feat: yyy
- fix: zzz

would bump the feature version as a feature commit was included.

**Reply by ga-paul-t** [reply]
I have just recently updated the logic within uplift to adopt a scanning approach to commits, instead of just defaulting to the latest commit. It doesn't support the use cases above just yet. But the intention is definitely to do so. At the moment uplift will scan for the first conventional commit message since the last release tag. It helps in scenarios where you have commit messages created through merges for example. But it doesn't help with those other use cases.

I am drafting a ticket to support these use cases as I believe it is needed in uplift.

I really appreciate the feedback

**Reply by ga-paul-t** [reply]
Tracking through issue: https://github.com/gembaadvantage/uplift/issues/195

**Reply by ga-paul-t** [reply]
@tomphp uplift currently impersonates the user who pushed a commit that triggered a release. In this instance, the commit could be one buried within a series of others. Without modifying this behaviour, uplift will impersonate the user whose commit triggers the largest semver bump. Would this cause you any issues? 

I have been thinking about improving this side of uplift as well. Toying with the idea of creating an Uplift Bot user. Also need to think about SCMs like GitLab that can enforce gpg signing on commits. Sorry, just thinking out loud here...

**Reply by tomphp** [reply]
I don't really have a preference over what name is on the release. Highest semver commit or bot are both fine for me 👍

**Comment by ga-paul-t** [comment]
@tomphp @gotson uplift will now scan all commits during a release and select the commit that generates the highest semver

**Reply by tomphp** [reply]
It seems to be working! Thank you!
